Feat/my actions

0xRowdy opened the pull request:
Created mobile and desktop ui for My Actions page.
Issue #14 

Desktop
<img width="1510" alt="Screenshot 2024-02-13 at 8 21 01 PM" src="https://github.com/VoiceDeck/app/assets/22626267/ff6b16dd-d3d8-45f1-8b28-520193e97c0b">

Mobile
<img width="1510" alt="Screenshot 2024-02-13 at 8 21 32 PM" src="https://github.com/VoiceDeck/app/assets/22626267/48950311-75a9-4bcd-adcd-95c143a89151">
<img width="1510" alt="Screenshot 2024-02-13 at 8 21 45 PM" src="https://github.com/VoiceDeck/app/assets/22626267/27664f89-63f3-44e2-98d4-dce40bdf5d56">
<img width="1510" alt="Screenshot 2024-02-13 at 8 21 50 PM" src="https://github.com/VoiceDeck/app/assets/22626267/678ccfb6-b482-411f-946b-f636e42c2aaa">

0xRowdy commented:
I'm realizing that the folder components/actions might not be clear. 
We could address this in a few ways. Have a components folder in the folder of the route, or I could rename the components to indicate the page they are associated with ex. actions-history, actions-history-card, etc. Open to other ideas.

0xRowdy commented on app/components/actions/history-card.tsx:
```diff
@@ -0,0 +1,80 @@
+} from "~/components/ui/card";
+import { Separator } from "~/components/ui/separator";
+import { formatCurrency, formatDate } from "~/lib/utils";
+import DynamicCategoryIcon from "../ui/dynamic-category-icon";
```
```suggestion
import DynamicCategoryIcon from "~/components/ui/dynamic-category-icon";
```

0xRowdy commented on app/components/actions/history-card.tsx:
```diff
@@ -0,0 +1,80 @@
+	CardTitle,
+} from "~/components/ui/card";
+import { Separator } from "~/components/ui/separator";
+import { formatCurrency, formatDate } from "~/lib/utils";
```
move lib imports above components

0xRowdy commented on app/components/actions/history-card.tsx:
```diff
@@ -0,0 +1,80 @@
+					/>
+					<div className="flex flex-col gap-3">
+						<CardTitle>{title}</CardTitle>
+						{/* TODO: Refactor to use DynamicCategoryIcon component from Beyonder */}
```
```suggestion
```

0xRowdy commented on app/components/actions/history-card.tsx:
```diff
@@ -0,0 +1,80 @@
+								className="rounded-3xl justify-between items-center"
+							>
+								<DynamicCategoryIcon category={category} />
+								{/* <Salad className="text-vd-orange-400 stroke-1 size-4 mr-1" /> */}
```
```suggestion
```

0xRowdy commented on app/components/actions/history-card.tsx:
```diff
@@ -0,0 +1,80 @@
+								{/* <Salad className="text-vd-orange-400 stroke-1 size-4 mr-1" /> */}
+								<p className="font-light text-xs ml-1">{category}</p>
+							</Badge>
+							{/* TODO: Refactor to use DynamicCategoryIcon component from Beyonder */}
```
```suggestion
```

0xRowdy commented on app/components/actions/history-card.tsx:
```diff
@@ -0,0 +1,80 @@
+								className="rounded-3xl justify-between items-center"
+							>
+								<DynamicCategoryIcon category="Location" />
+								{/* <MapPin className="text-vd-orange-400 stroke-1 size-4 mr-1" /> */}
```
```suggestion
```

0xRowdy commented on app/components/actions/summary.tsx:
```diff
@@ -0,0 +1,62 @@
+				<CardContent>
+					{/* TODO: Make Dynamic based on user */}
+					<div className="grid grid-cols-1 md:grid-cols-2 gap-2 md:gap-4">
+						{/* TODO: Refactor to use DynamicCategoryIcon component from Beyonder */}
```
```suggestion
```

CJ-Rose commented:
> I'm realizing that the folder components/actions might not be clear.
We could address this in a few ways. Have a components folder in the folder of the route, or I could rename the components to indicate the page they are associated with ex. actions-history, actions-history-card, etc. Open to other ideas.

i find the the current structure to be clear. can also have `components/reports` and `components/report-detail` folders to hold the components used in those routes.
removing the folder and renaming files `actions-history`, `reports-voicedeck-stats`, etc could be good too, as the files would appear grouped together alphabetically.
it makes more sense to me to keep all the components inside the components folder (along with `global` and `ui`) rather than splitting these off into the related route folders.

0xRowdy commented on app/routes/_index.tsx:
```diff
@@ -36,7 +36,7 @@ import { siteConfig } from "~/config/site";
 import { fetchReports } from "../impact-reports.server";
 
-const iconComponents: { [key: string]: LucideIcon } = {
+export const iconComponents: { [key: string]: LucideIcon } = {
```
Done

0xRowdy commented on app/components/actions/summary.tsx:
```diff
@@ -0,0 +1,61 @@
+import { Badge } from "~/components/ui/badge";
+import { Card, CardContent, CardHeader, CardTitle } from "~/components/ui/card";
+import { iconComponents } from "~/routes/_index";
```
Good catch! Done

thebeyondr commented:
> We could address this in a few ways. Have a components folder in the folder of the route, or I could rename the components to indicate the page they are associated with ex. actions-history, actions-history-card, etc. Open to other ideas.

 `component/my-actions` might be clearer. 
 
 However, there are even more possibilities with Remix. Let's examine one: **Colocation**
 
 ```
 routes/
├── [...]
├── my-actions/
│   ├── index.tsx # /my-actions 
│   ├── history-card.tsx
│   ├── summary.tsx
```

thebeyondr commented on app/lib/utils.ts:
```diff
@@ -4,3 +4,21 @@ import { twMerge } from "tailwind-merge";
 	return twMerge(clsx(inputs));
 }
+
+export function formatDate(date: Date) {
```
noice

thebeyondr reviewed (changes requested):
Looking good so far, but I can't approve yet because: 

- [x] small css typo (600 uses 400 color). 
- [ ] Empty state unaccounted for
- [x] Some cards have wrong bg color (see figma)
- [ ] Some bento areas and cards have shadows where they should not (see figma)

0xRowdy commented:
> > We could address this in a few ways. Have a components folder in the folder of the route, or I could rename the components to indicate the page they are associated with ex. actions-history, actions-history-card, etc. Open to other ideas.
> 
> `components/my-actions` might be clearer.
> 
> However, there are even more possibilities with Remix. Let's examine one: **Colocation**
> 
> ```
> routes/
> ├── [...]
> ├── my-actions/
> │   ├── index.tsx # /my-actions 
> │   ├── history-card.tsx
> │   ├── summary.tsx
> ```

Yeah, I like this approach. Is that a good home for components or is that generally used for view/routes? I might need to do a little bit of re-factoring of this route due to the default state being the un-authenticated state.

thebeyondr commented:
> Yeah, I like this approach. Is that a good home for components or is that generally used for view/routes? I might need to do a little bit of re-factoring of this route due to the default state being the un-authenticated state.

Here's the [direct reference in Remix v2](https://remix.run/docs/en/main/file-conventions/routes#folders-for-organization). I got some naming things wrong. If the user is unauthenticated then they wouldn't have a `My Actions` link. and if they try to access the route they should be redirected to '/'

0xRowdy commented:
> > Yeah, I like this approach. Is that a good home for components or is that generally used for view/routes? I might need to do a little bit of re-factoring of this route due to the default state being the un-authenticated state.
> 
> Here's the [direct reference in Remix v2](https://remix.run/docs/en/main/file-conventions/routes#folders-for-organization). I got some naming things wrong.
> 
> If the user is unauthenticated then they wouldn't have a `My Actions` link in nav. If they try to access the route manually they should be redirected to '/'

Yeah that is what I would expect. Currently, the design looks like it would ask for auth if you try to navigate to my-actions. We could add connect wallet to the header and not show the My Actions Link unless the user has signed in with their wallet. That might kinda render the design useless unless we did some route handling and made the my-actions unauthenticated state its own route called auth, log-in or sign-in.

We've got some options here.

0xRowdy commented on tailwind.config.js:
```diff
@@ -19,9 +19,10 @@ module.exports = {
       colors: {
         "vd-beige": {
-          200: "rgb(var(--vd-beige-200) / <alpha-value>)",
+          600: "rgb(var(--vd-beige-400) / <alpha-value>)",
```
done

0xRowdy commented:
> Looking good so far, but I can't approve yet because:
> 
> * [ ]  small css typo (600 uses 400 color).
> * [ ]  Empty state unaccounted for
> * [x]  Some cards have wrong bg color (see figma)
> * [x]  Some bento areas and cards have shadows where they should not (see figma)

Can you help me understand?

My Contribution card = --vd-blue-200: 194 214 223;

<img width="1125" alt="Screenshot 2024-02-15 at 5 20 49 PM" src="https://github.com/VoiceDeck/app/assets/22626267/0f4b028c-9b7b-4445-828c-39e3fe0209c9">

#of Reports card = --vd-beige-300: 241 224 215;
<img width="1111" alt="Screenshot 2024-02-15 at 5 40 55 PM" src="https://github.com/VoiceDeck/app/assets/22626267/c462e484-224f-4ac7-9fe9-37f9f862bfb4">


> Looking good so far, but I can't approve yet because:
> 
> * [ ]  small css typo (600 uses 400 color).
> * [ ]  Empty state unaccounted for
> * [ ]  Some cards have wrong bg color (see figma)
> * [ ]  Some bento areas and cards have shadows where they should not (see figma)

I don't follow you on Empty state unaccounted for. if you are referring to history it has an empty state when !history. I originally added a check for the history object like {history && history.map => ... but the linter complained and had me update it to {history?.map

Lmk if this is what you were talking about with empty state or something else.

thebeyondr commented:
> Lmk if this is what you were talking about with empty state or something else.

Yes I meant History empty state. Seems I missed it my bad. I refactored it. Not sure why you're not using default exports for components though but I changed History to be one.

0xRowdy commented:
> > Lmk if this is what you were talking about with empty state or something else.
> 
> Yes I meant History empty state. Seems I missed it my bad. I refactored it. Not sure why you're not using default exports for components though but I changed History to be one.

I generae

> > Lmk if this is what you were talking about with empty state or something else.
> 
> Yes I meant History empty state. Seems I missed it my bad. I refactored it. Not sure why you're not using default exports for components though but I changed History to be one.

My general rule of thumb is to use default exports for pages and named exports for components. I've had several projects and repos that enforce only named exports. It used to be more of an issue with tree shaking and with IDEs offering better autocomplete and import suggestions with named exports.

thebeyondr commented:
> My general rule of thumb is to use default exports for pages and named exports for components. I've had several projects and repos that enforce only named exports. It used to be more of an issue with tree shaking and with IDEs offering better autocomplete and import suggestions with named exports.

makes sense, thanks for the explaining 🫡

0xRowdy commented:
@thebeyondr Anything else here? I got all the requested changes but you still might need to re-approve.
